### PR TITLE
Adding powershell script for Module 4

### PR DESCRIPTION
**Pull request by rajeshdewan**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: CLI.ps1 [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+if ($ispoetryinstalled) {poetry --version} else { (Invoke-WebRequest -Uri https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py -UseBasicParsing).Content | python }
+
+#To get started you need Poetry''s bin directory (%USERPROFILE%\.poetry\bin) in your `PATH`
+
+$env:PATH += ";%USERPROFILE%\.poetry\bin;"
+
+#Install application dependencies
+cd C:\DEVOPS-Copy\DevOps-Course-Starter # change dir
+#git init
+git clone https://github.com/CorndelWithSoftwire/DevOps-Course-Starter.git
```
Comment (hugh-emerson) [inline CLI.ps1:36]: If you want to clone it into the current directory you need a dot at the end to specify that. The default behaviour is to create a subdirectory with the repo's name, so you would need to cd into it before the next command (poetry install)

```
git clone https://github.com/CorndelWithSoftwire/DevOps-Course-Starter.git .
```

Note that cloning into an existing directory is only allowed when the directory is empty.

File: CLI.ps1 [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+$ispythoninstalled = python --version
+if ($ispythoninstalled) { python --version } else { Invoke-WebRequest -Uri "https://www.python.org/ftp/python/3.7.0/python-3.7.0.exe" -OutFile "C:\DEVOPS-Copy\DevOps-Course-Starter/python-3.7.0.exe" }
+
+
+#command to install Python
+C:\DEVOPS-Copy\DevOps-Course-Starter/python-3.7.0.exe /quiet InstallAllUsers=0 PrependPath=1 Include_test=0
+
+##update the path 
+ENV:PATH="$ENV:PATH;C:\Users\Rajesh\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\Scripts"
```
Comment (hugh-emerson) [inline CLI.ps1:19]: Shouldn't it be `$ENV:PATH="...`?

Also:
- This is very specific to your machine, could you make this general?
- This only updates the PATH environment variable for the current process. If you opened up a new powershell terminal on this machine, it won't know about this. To update your profile so the value is available for all future shells, there's a more complicated command along the lines of:
```
[System.Environment]::SetEnvironmentVariable('ENV_VAR_NAME', 'env var value', [System.EnvironmentVariableTarget]::User)
```
depending on the exact behaviour you want

**Review by hugh-emerson — APPROVED** [review]
You have the right sort of idea for what a provisioning script is aiming to achieve but see my comments below for some thoughts.

Merge when you're ready

File: CLI.ps1 (hunk 1 above)
Comment (rajeshdewan) [reply CLI.ps1:36]: OK, noted. Thanks

File: CLI.ps1 (hunk 2 above)
Comment (rajeshdewan) [reply CLI.ps1:19]: Yes, what you pointed out is correct. For the comment on PATH , I believe this is what I was trying to play with at the time which somehow managed to clear out variables from my settings.

**Pull request merged**